v2/spiffetls/dial.go

```diff
@@ -25,10 +36,63 @@ func (c *Conn) Close() error {
 //
 // This is the same as DialWithMode using the MTLSClient mode.
 func Dial(ctx context.Context, network, addr string, authorizer tlsconfig.Authorizer, options ...DialOption) (*Conn, error) {
-	panic("not implemented")
+	return DialWithMode(ctx, network, addr, MTLSClient(authorizer), options...)
 }
 
 // DialWithMode creates a TLS connection using the specified mode.
 func DialWithMode(ctx context.Context, network, addr string, mode DialMode, options ...DialOption) (*Conn, error) {
-	panic("not implemented")
+	m := mode.get()
+	var err error
+	var source *workloadapi.X509Source
+	if m.source != nil {
+		source = m.source
+	} else {
+		source, err = workloadapi.NewX509Source(ctx, m.options...)
```

Comment:
The source needs to be closed (via defer, preferably) if the function fails after this point.